src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -318,6 +317,21 @@ private JsonConverter GetConverterFromAttribute(JsonConverterAttribute converter
             return converter;
         }
 
+        internal bool TryGetDefaultSimpleConverter(Type typeToConvert, [NotNullWhen(true)] out JsonConverter? converter)
+        {
+            if (_context == null && // For consistency do not return any default converters for
+                                    // options instances linked to a JsonSerializerContext,
+                                    // even if the default converters might have been rooted.
```

Comment:
Good place to start from, enforcing consistency. I'd be opening to loosening it in the future if there's a compelling scenario.

Reply:
I think we should keep as-is: otherwise you would get different behaviour depending on whether the current process has invoked the reflection-based serializer!